src/inc/til/ticket_lock.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "atomic.h"
+
+namespace til
+{
+    // The use of alignas(std::hardware_destructive_interference_size) with this class is highly suggested.
+    struct ticket_lock
+    {
+        void lock() noexcept
+        {
+            const auto ticket = _next_ticket.fetch_add(1, std::memory_order_relaxed);
+
+            for (;;)
+            {
+                const auto current = _now_serving.load(std::memory_order_acquire);
+                if (current == ticket)
```

Comment:
there is provably no case where now_serving will jump multiple steps above the expected ticket value, correct?

Reply:
It is possible, if someone where to call `unlock()` multiple times.
But if you don't it works exactly like the Wikipedia article describes the algorithm.